docs/find-lost-trn-integration.md

```diff
@@ -0,0 +1,72 @@
+# Find a lost TRN integration with the authorization server
+
+When the `qualified_teacher` scope is requested by a client, the authorization server needs to integrate with the
+Find a lost TRN service in order to resolve a DQT contact record (via a TRN). The details of how data is handed over
+from the authorization server to Find and back again are listed below.
+
+
+```mermaid
+sequenceDiagram
+    participant client
+    participant authz_server
+    participant find_lost_trn
+    participant dqt_api
+    client->>authz_server: redirects user to the authorization endpoint with 'qualified_teacher' scope
+    authz_server->>authz_server: prompts user for email
+    alt existing user
+        authz_server->dqt_api: lookup DQT contact record with matching teacher_id
+    end
+    alt not existing user or lookup failed
+        authz_server->>find_lost_trn: redirects user with a callback URL + additional context
+        find_lost_trn->dqt_api: resolve TRN
+        find_lost_trn->>authz_server: redirects user to callback URL with JWT containing TRN
+        authz_server->dqt_api: persist the teacher_id against the DQT contact record
+    end
+    authz_server->>client: redirects user to the client with id_token containing email + TRN claims
+```
+
+
+## Handover from authorization server to Find a lost TRN
+
+When the authorization server needs a TRN for the user it redirects to the Find a lost TRN service.
+This redirect includes some additional context, specified as query parameters:
+
+| Query parameter | Remarks |
+| --- | --- |
+| email | The verified email address as captured by the authorization server. This allows Find a lost TRN to skip asking the user for their email address again. |
+| redirect_uri | This is the callback URL on the authorization server that Find should redirect the user to once it has resolved a TRN. |
+| client_title | The name of the client that initiated the authorization journey. This enables 'branding' both the the authorization server and Find a lost TRN such that the user perceives the journey as a single service e.g. 'Register for a National Professional Qualification'. |
+| sig | This is a signed hash of the previous query parameters using a pre-shared key. |
+
+### Context signature query parameter
```

Comment:
An alternative to this would be to use a JWT, just like in the handover back from Find. I opted against that here since most of the context is not claims about the user so using a JWT felt a little smelly.